apps/demos/Demos/Splitter/Overview/React/PaneContent.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useMemo } from 'react';
+
+export default function PaneContent(paneName: string) {
+  const dimensionOptions = new Set(['size', 'minSize', 'maxSize'])
+
+  return function template(data) {
+    const getPaneState = useMemo(() => {  
+      if (data.resizable !== false && !data.collapsible) {
+        return 'Resizable only';
+      }
+      return `${data.resizable ? 'Resizable' : 'Non-resizable'} and ${data.collapsible ? 'collapsible' : 'non-collapsible'}`;
+    }, [data.resizable, data.collapsible]);
+
+    const filterDimensionOptions = useMemo(() => Object.entries(data)
+      .filter(([key]) => dimensionOptions.has(key))
+      .map(([key, value]) => ({key, value})),
+    [data]);
```

Comment:
I think that data is always different here => `useMemo` only takes the memory but does not optimize anything

Reply:
[a91162c](https://github.com/DevExpress/DevExtreme/pull/26972/commits/a91162c87227d06b8645ebae24f59cb7473ae8ad)